Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,1195 @@
+            # Make sure they're not all in the same spot and in the 0..1 range:
+            # Note: This does not prevent overlapping UVs or faces with no area but that's up to the user to fix:
+            import numpy as np
+            uvOutOfBounds = False
+            for i in range(len(obj.data.uv_layers)):
+                uv_map = obj.data.uv_layers[i]
+                if uv_map.active:
+                    nl = len(obj.data.loops)
+                    uv_verts = np.zeros(nl * 2)
+                    uv_map.data.foreach_get("uv", uv_verts)
+                    uniqueUVs = np.unique(uv_verts.round(decimals=4))
+                    greater = np.any(np.greater(uniqueUVs, 1))
+                    if greater:
+                        uvOutOfBounds = True
+                        break
+                    less = np.any(np.less(uniqueUVs, 0))
+                    if less:
+                        uvOutOfBounds = True
+                        break
+                    break
+
+            if uvOutOfBounds:
+                if bakeSettings.generate_uvs == 'OFF':
+                    raise Exception(
+                        f"Object: '{obj.name}' has no usable UVs. Automatically generating UVs is disabled, so this object cannot be baked!")
+
+                msb_log("UVs are not in 0..1 range for baking, generating new UVs.")
+                generateUVs = True
+
+        if generateUVs:
+            msb_log(f"Auto generating UVs for object: '{obj.name}'.")
+
+            bakeUVLayer = obj.data.uv_layers.new(name="Baked")
+            obj.data.uv_layers.active = bakeUVLayer
+            # Store uv override on the mesh, we'll need this later to delete all other UV layers,
+            # otherwise the baked images will use the wrong UVs!
+            obj.data[UV_OVERRIDE] = bakeUVLayer.name
+
+            if context.object.mode == 'OBJECT':
+                bpy.ops.object.mode_set(mode='EDIT')
+
+            bpy.ops.mesh.select_mode(use_extend=False, use_expand=False, type='VERT')
+            bpy.ops.mesh.select_all(action='SELECT')
+            bpy.ops.mesh.select_linked(delimit={'SEAM'})
+            bpy.ops.uv.smart_project(island_margin=0.01, scale_to_bounds=True)
+            bpy.ops.uv.pack_islands(rotate=True, margin=0.001)
+
+        if UV_OVERRIDE in obj.data:
+            obj.data.uv_layers.active = obj.data.uv_layers[obj.data[UV_OVERRIDE]]
+        else:
+            obj.data.uv_layers.active = obj.data.uv_layers[0]
+
+        if obj.mode == 'EDIT':
+            bpy.ops.object.mode_set(mode='OBJECT')
+
+    def getNodeYLocation(self, node):
+        location = node.location[1]
+        if node.parent is not None:
+            location += self.getNodeYLocation(node.parent)
+        return location
+
+    def prepareMaterial(self, context, obj, bsdf, mat, canBakeBSDF):
+        # Make material copy if this is not a copy already:
+        if ORIGINAL_MATERIAL not in mat:
```

Comment:
Can you invert these if statements to reduce the nesting?

```
if ORIGINAL_MATERIAL in mat:
   return mat

... Do logic stuff ...
```

Reply:
Done